woocommerce/woocommerce
# Improve rendering of Order Attribution inputs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



- [Expose public method to render OA input fields](https://github.com/woocommerce/woocommerce/commit/703635e8017142823871e5723f2a67f17d3f503d) 
- [Render OA inputs on woocommerce_checkout_after_customer_details](https://github.com/woocommerce/woocommerce/commit/ca8a6aa6c9f465692a6ebd462046daaf8a0e719a) 
   in case the checkout form does not contain `woocommerce_checkout_shipping`, `woocommerce_after_order_notes`.

Addresses part of https://github.com/woocommerce/woocommerce/issues/44159



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Edit your theme/template to remove `woocommerce_checkout_shipping`
   - go to _Tools > Plugin File Editor > Select plugin to edit:  "WooCommerce" > Plugin Files "templates/checkout/form-checkout.php"_  
      ```
      /wp-admin/plugin-editor.php?file=woocommerce%2Ftemplates%2Fcheckout%2Fform-checkout.php&plugin=woocommerce%2Fwoocommerce.php
      ```
   - remove the line with `<?php do_action( 'woocommerce_checkout_shipping' ); ?>`
2. Open the classic (non-Block) checkout page
   - There should be no errors
4. Open browser dev tools, call: `document.querySelectorAll( `input[name^="${wc_order_attribution.params.prefix}"]` )`
   - There should be 13 elements
      ![image](https://github.com/woocommerce/woocommerce/assets/17435/c1825319-691a-4909-b07b-00dd17c907c8)

1. Edit your theme/template to remove `woocommerce_checkout_after_customer_details`
2. Open the classic (non-Block) checkout page
   - There should be no errors
4. Open browser dev tools, call: `document.querySelectorAll( `input[name^="${wc_order_attribution.params.prefix}"]` )`
   - There should empty `NodeList[]`




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Move rendering of Order Attribution inputs to `woocommerce_checkout_after_customer_details`. Expose a method for custom theming.

#### Comment 

</details>
